JUnit 5: Test do not run, but silently succeed if not all classes are packaged

I noticed, that when adding additional functionality to tests, and forgetting to add the class to the `@Deployment` method,
the tests stop running, but are being reported as successful by the Failsafe plugin.

This is especially insidious because tests appear to succeed, but are not even run.
The only way I even noticed this is by test coverage mysteriously going down when it should be going up.

I wouldn't be surprised if this affects the Jakarta EE TCK in some way as well.

Reproducer: https://github.com/flowlogix/arq-reproducers
PR: (maybe?) TBD

With the PR, the reproducer fails as expected:
```
ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.156 s <<< FAILURE! -- in com.arq.test.StarterIT
[ERROR] com.arq.test.StarterIT.sanityCheck -- Time elapsed: 0.067 s <<< ERROR!
java.lang.NoClassDefFoundError: org/slf4j/Logger
	at java.base/java.lang.Class.getDeclaredFields0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredFields(Class.java:3545)
	at java.base/java.lang.Class.getDeclaredFields(Class.java:2601)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
Caused by: java.lang.ClassNotFoundException: org.slf4j.Logger
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1876)
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1712)
	... 4 more
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   StarterIT.sanityCheck » NoClassDefFound org/slf4j/Logger
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

--- a/junit5/container/src/main/java/org/jboss/arquillian/junit5/container/JUnitJupiterTestRunner.java
+++ b/junit5/container/src/main/java/org/jboss/arquillian/junit5/container/JUnitJupiterTestRunner.java
@@ -68,6 +68,7 @@ public class JUnitJupiterTestRunner implements TestRunner {
 
     private static class ArquillianTestMethodExecutionListener implements TestExecutionListener {
         private final Map<String, Throwable> exceptions = new HashMap<>();
+        private Throwable fatalError;
 
         public void executionSkipped(TestIdentifier testIdentifier, String reason) {
             exceptions.put(testIdentifier.getUniqueId(), new TestAbortedException(reason));
@@ -77,6 +78,9 @@ public class JUnitJupiterTestRunner implements TestRunner {
             TestExecutionResult.Status status = testExecutionResult.getStatus();
 
             if (!testIdentifier.isTest()) {
+                if (status != TestExecutionResult.Status.SUCCESSFUL) {
+                    fatalError = testExecutionResult.getThrowable().orElseGet(() -> new Exception("Failed"));
+                }
                 return;
             }
             switch (status) {
@@ -98,6 +102,9 @@ public class JUnitJupiterTestRunner implements TestRunner {
         }
 
         private TestResult getTestResult() {
+            if (fatalError != null) {
+                return TestResult.failed(fatalError);
+            }
             return exceptions.isEmpty() ? TestResult.passed() : TestResult.failed(new IdentifiedTestException(exceptions));
         }
     }

